src/wwwroot/template-editor/adapterService.js

```diff
@@ -339,41 +352,22 @@
      */
     function getSettings(idCampaign, idTemplate) {
       traceAdapterCall(getSettings, arguments);
+      var url = "/#/"
+      if (idCampaign != 0)
+      {
+        url = "/template-editor/?idCampaign=" + idCampaign;
+      }
+      if (idTemplate != 0)
+      {
+        url = "/template-editor/?idTemplate=" + idTemplate;
+      }
```

Comment:
This redirectUrl is not being used by back button, sorry, this is used to handle a specific match from response html in a httpInterceptor:
```js

    function response(response) {
      if (typeof response.data === 'string' && (response.data.indexOf('<title>Login - Doppler Email Marketing</title>') !== -1 ||
        response.data.indexOf('<title>Ingresa - Doppler Email Marketing</title>') !== -1)) {

        //We create a temporary internal message to emulate a 401 unauthorized message
        var error = {
          status: 401,
          message: 'Unauthorized'
        };

        showModalAndRedirect();

        return $q.reject(error);
      }

      return response;
    }
```

this interceptor calls to:
```js
    function showModalAndRedirect() {
     var settingsPromise;
     var changesManager;

     var modalOptions = {
        title: $translate.instant('errors.server_error'),
        message: $translate.instant('errors.saving_changes'),
        buttons: [{
          label: $translate.instant('accept'),
          class: 'button--primary'
        }]
      };

      return modalService.showModal(modalOptions).then(function() {
        settingsPromise = $injector.get('settingsService').getSettings();
        changesManager = $injector.get('changesManager');

        return settingsPromise.then(function(settings) {
          changesManager.setUnsavedChangesCampaign(false);
          $window.location.href = settings.redirectUrl;
        });
      });
    }
```

Here you will found the redirection using this json attribute:
`$window.location.href = settings.redirectUrl`

This is the only one place where is used. I checked the doppler backend and this attribute is being setted depending the value of idCampaigns and idTemplates, but the MSEditor app is handling this behavior too.

MSEditor catch the value of idCampaign or idTemplates:
```js
if(isNaN(idCampaign) && isNaN(idTemplate) || idCampaign && idTemplate) {
              return $q.reject('Url should have an idCampaign or an idTemplate as in: "/?idCampaign=1" or "/?idTemplate=1"');
            }
```

Then, the behavior on the interceptor is never used. I think maybe this can be a security patch or something like that.
I found in google the <title> being listened in the interceptor and match with **https://cm.fromdoppler.com/**

I really think that we don't need this setting, but by the moment I will handle locally with the same logic that the backend in doppler used.

Doppler backend getSettings method:
```c#
if ((idCampaign == 0) && (idTemplate == 0))
{
	dtoEditorSetting.RedirectURL = "/SignIn";
}
else
{
	if (idCampaign != 0)
	{
                // More code here
		dtoEditorSetting.RedirectURL = "/MSEditor/Editor?idCampaign=" + idCampaign.ToString();
	}
	if (idTemplate != 0)
	{
                // More code here
		dtoEditorSetting.RedirectURL = "/MSEditor/Editor?idTemplate=" + idTemplate.ToString();
	}
}
```

Sorry for the big comment, let me know your thoughts.

Reply:
I think that I am understanding. It seems that backend work is not necessary.
Lets talk about it when you have time.